src/AggregateFunctions/AggregateFunctionUniq.h

```diff
@@ -162,17 +173,29 @@ template <typename T> struct AggregateFunctionUniqTraits
 };
 
 
-/** The structure for the delegation work to add one element to the `uniq` aggregate functions.
+/** The structure for the delegation work to add elements to the `uniq` aggregate functions.
   * Used for partial specialization to add strings.
   */
-template <typename T, typename Data>
-struct OneAdder
+template <typename T, typename Data, bool is_variadic = false, bool is_exact = false, bool argument_is_tuple = false>
```

Comment:
I had an idea that probably we can put this 3 flags into `Data`.
Probably it will make a code a bit more readable. (If it is possible to do).

And maybe we can do it with `is_able_to_parallelize_merge` flag as well to those Data which support it, like
```
if (settings->max_threads > 1)
            return createAggregateFunctionUniq<
                ..., AggregateFunctionUniqExactData<..., true /* is_able_to_parallelize_merge>
else
            return createAggregateFunctionUniq<
                ..., AggregateFunctionUniqExactData<..., false /* is_able_to_parallelize_merge>
```

Reply:
also thought about that, looks more accurate